### ISSUE TITLE
Validate binding behavior not working

### ISSUE THREAD

**Issue by MaximBalaganskiy** [grofit/treacherous-aurelia#4]
Validate binding behavior not working unless I expose `validationGroup` property on a view model. Any other variable name does not work. There are two problems I observe:
- `validate.bind` is called before `validation-group.bind`
- even if the sequence was correct, `validation-group.bind` adds `validationGroup` property to the `overrideContext` while `validate.bind` looks for it on `bindingContext`

**Comment by grofit** [comment]
will look into it, as it works fine in the examples (which are the test harness), I remember that there are some quirks around how aurelia deals with binding scopes in child areas, like for example if you were within a repeat I believe you want the parent of the child and then you can only get the binding context not the override context, but one also populates the other in some scenarios I think.

It is odd though that validate is called before validation-group, can you make a simple repro on cloud9 or something?

**Comment by MaximBalaganskiy** [comment]
Here is the [gist](https://gist.run/?id=89e478227240a83c0ce4765526078846) I created.
Please look at the console log for debug messages. 

I excluded `treacherous-aurelia` package from the bundle and included it into gist so you can quickly tweak it.

**Comment by MaximBalaganskiy** [comment]
Further on this, I think one of the fixes could be to just store the validation group property name in an attribute and search for it on parent elements in the binding behavior.

main.html

```
<template>
    <form validation-group="valGroup"> <!-- please note, no binding here, it's just an attribute -->
        <input value.bind="stringValue & validate" />
        <button type=submit>Submit</button>
    </form>
</template>
```

validate-binding-behaviour.js

```
// replace this this.validationGroup = overrideContext.bindingContext.validationGroup;
overrideContext.bindingContext[binding.target.parentElement.attributes["validation-group"].value]
```

Now, this snippet is only checking the direct parent, ideally i would traverse the whole tree and if no validation group attribute is found then use default `validationGroup` value

[Here](https://gist.run/?id=644ef1af09c6f291067eb037137084b6) is the proposed fix in action

**Comment by grofit** [comment]
Hey, you able to get on gitter to have a quick chat about this just so I am clear on the issue? as it worries me that for some reason aurelia is calling the child binding before the parent one, so am not sure if this is a problem with aurelia in some way, as I don't want to second guess their binding system but also if you end up with a complex stack it wont always be the parent element that has the validation group, it may be 5 elements up the tree so you cannot just assume its the parent element, but again assuming aurelia binds the way it should (and does in the example/previous versions) it should all work.

**Comment by MaximBalaganskiy** [comment]
Lets hear what the team says
http://stackoverflow.com/questions/38516504/aurelia-bindings-processing-order

**Comment by grofit** [comment]
Is this all working for you now by the way? as if so can we get this closed.

**Comment by MaximBalaganskiy** [comment]
No, that SO answer is the best I have and it says that we cannot rely on processing order. Binding behavior will not work with properties which are not stored in `validationGroup` variable.
Moreover, storing the current validation group in a variable will not work with dynamically added elements because they will always grab the latest stored validation group. For now, to correctly use treacherous with multiple groups I will have to use my own custom attribute which walks up the tree to find the right group.

**Comment by grofit** [comment]
Ok well the ultimate fix will have to remain with someone who has the time to chase down the aurelia team to find out how best to use the binding context, as far as a quick fix I can provide a validation group argument to the validation bindings to allow you to stipulate one if it is not in scope if that would work for you.

**Comment by grofit** [comment]
As I was chatting to Jason about this, I think `bind` fires first then `attached` so I _think_ it may work if some of the logic is moved out of the `bind` call into the `attached` call.

If you have a moment at some point I have made a quick gist to see if this would work for you:
https://gist.github.com/grofit/27fc02a9fbf1a08aa02292f9d223a5aa

if you take that and put that into `validate-property-attribute` in your treacherous-aurelia module folder (be it jspm or npm) and then try running and see if that solves this problem. For some reason it doesnt happen on the example app or other use cases I have tried so I cannot replicate.
